### PR TITLE
Add io.github.andreibachim.shortcut

### PR DESCRIPTION
**Pull request by andreibachim**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)* 

>   The application requires full access to the host file system because it needs to locate files anywhere on the host. When using only read-only permission, the FileChooser portal returns document portal fuse filesystem path, which is not suitable to be saved in .desktop files and can even confuse less technical users.

- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/andreibachim/shortcut
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* - not relevant

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: io.github.andreibachim.shortcut.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  '--socket=fallback-x11',
+  '--share=ipc',
+  '--device=dri',
+  "--filesystem=host",
+]
+modules:
+  - name: shortcut
+    buildsystem: simple
+    build-commands:
+      - install -D target/release/shortcut /app/bin/shortcut
```
Comment (hfiguiere) [inline io.github.andreibachim.shortcut.yml:17]: how is this build?

File: flathub.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "only-arches": [
+        "x86_64"
```
[the remark this answers is not in the document]
Comment (andreibachim) [reply flathub.json:3]: The release zip only contains the binary for x86_64 for now. I need to cross-compile for aarch64 and test before adding support for that target.

File: io.github.andreibachim.shortcut.yml (hunk 1 above)
Comment (andreibachim) [reply io.github.andreibachim.shortcut.yml:17]: Currently, it is built locally using [this script](https://github.com/andreibachim/shortcut/blob/main/scripts/release.sh). The plan is to move to github actions sometime in the near future.

Comment (hfiguiere) [reply io.github.andreibachim.shortcut.yml:17]: you should build it here on flathub. Like most are done.

Comment (andreibachim) [reply io.github.andreibachim.shortcut.yml:28]: I have updated the manifest in order to build the binary in place.

**Comment by andreibachim** [comment]
bot, build io.github.andreibachim.shortcut

**Comment by andreibachim** [comment]
bot, build io.github.andreibachim.shortcut

File: flathub.json
Comment (hfiguiere) [inline flathub.json:1]: new you can remove this file.

**Comment by hfiguiere** [comment]
bot, build io.github.andreibachim.shortcut

**Comment by andreibachim** [comment]
Hi, do I need to do something specific to move this forward?

**Comment by andreibachim** [comment]
bot, build io.github.andreibachim.shortcut

**Comment by andreibachim** [comment]
bot, build io.github.andreibachim.shortcut

**Comment by andreibachim** [comment]
> The app should be changed to use DynamicLauncher portal for creating launchers.

I'm looking into it. Thank you for the suggestion.

**Comment by andreibachim** [comment]
Hi,

So I looked into DynamicLauncher portal, and from what I can tell, it could be used for this app as it currently stands.

However, there are some drawbacks that go against my vision/future plans for this app.

1) From what I can understand from the docs (and tested with Epiphany), created launchers are deleted when you uninstall the flatpak that created them. Which is not what I want to happen. 

2) When the portal is used in a sandbox the `Exec` value is overriden, which is not ok, because I do not want the .desktop file to be tied in with Shortcut in any way. 

3) I plan to gradually include more options to the app, like editing existing .desktop files, a expert mode with full support for the freedesktop spec for desktop entries, saving .desktop files to user an user selected location for usage in other scenarios (not only installing locally). 

Can this move forward without implementing this suggestion?

Thank you

**Comment by andreibachim** [comment]
Hi, @barthalion ,  any feedback/input on my [comment](https://github.com/flathub/flathub/pull/4437#issuecomment-1694763337) above?

**Pull request closed without merging**